packages/chevrotain/src/parse/parser/parser.ts

```diff
@@ -166,7 +167,10 @@ export class Parser {
           this.enableRecording()
           // Building the GAST
           forEach(this.definedRulesNames, (currRuleName) => {
-            const wrappedRule = this[currRuleName]
+            const wrappedRule = (this as any)[currRuleName] as ParserMethod<
```

Comment:
sometimes the logic is too dynamic for TypeScript and fighting with Types deducts rather then adds.
But no need to change this... its subjective...